README.md

```diff
@@ -1,99 +1,128 @@
+
+	p.AddOptions(dry_run, max)
+
+	// Parse all available program arguments (except for the program path).
+	if ns, leftovers, err := p.Parse(os.Args[1:]...); err != nil {
+		switch err.(type) {
+		case argparse.ShowHelpErr, argparse.ShowVersionErr:
+			return
+		default:
+			fmt.Println(err, "\n")
+			p.ShowHelp()
+		}
+	} else {
+		name := ns.String("name")
+		upper := ns.String("upper") == "true"
+
+		if upper == true {
+			name = strings.ToUpper(name)
+		}
+
+		fmt.Printf("Hello, %s!\n", name)
+		if len(leftovers) > 0 
+		{
+            // Print slice of unexpected arguments passed to the program.
+			fmt.Println("\nUnused args:", leftovers) 
+		}
+	}
 }
 ```
 
-## Adding Flags
-Now that we have a parser, lets add two flags and some program functionallity. If
-the user provides a `--foo` flag, we will output the string: `foobar!`. In addition,
-we will output zero by default, or whatever number the user provides with the `-i I` flag.
+A parser can provide a version number, automatically add `-h --help` and `-v --version`
+options, and is used to contain a collection of Options.
 
-```go
-package main
+## Options
+Options are expected parameters to your parser. `argparse.Option` is the base
+struct for parseable options. 
 
-import (
-    "fmt"
-    "os"
+Options can represent a variety of input types, but are always serialized to
+a `string`. Options may be required or not, may be positional or not, can have
+a variable number of parameters, and can operate in a variety of manners.
 
-    "github.com/parg"
-)
+That said, there are two common types of Options. As such, help functions are 
+included to make it easier to create them: Flags and Positional Arguments.
 
-func main() {
-    p := parg.NewParser("Output text based on user input")
-    p.AddHelp() // Enable `--help` & `-h` to display usage text to the user.
-
-    foo := parg.NewFlag("foo", "Enable foobar text output")
-    num := parg.NewFlag("i", "Set desired integer number to output").Nargs("1").Action(parg.Store).Default(0)
-
-    p.AddFlag(foo).AddFlag(num)
-
-    // Parse all available program arguments (except for the program path).
-    if err := p.Parse(os.Args[1:]...); err != nil {
-        // An error occurred? Print it out, and display the help text!
-        fmt.Println(err)
-        p.ShowHelp()
-    } else {
-        if p.Values["foo"] == true {
-            fmt.Println("foobar!")
-        }
-        fmt.Println(p.Values["i"])
-    }
-}
+### Flags
+Flags represent non-positional, boolean arguments. These Options are `"false"` 
+by default, will utilize the `argparse.StoreTrue` action. They do not any parameters. You can create a new flag using:
+
+```go
+// Create a -d --default flag, called "use-default", with a provided description.
+use_default := argparse.NewFlag("d default", "use-default", "Enable the default mode")
+```
+
+### Arguments
+Arguments represent positional, 1-parameter options. They will store the
```

Comment:
Again, this description is not as clear as it should be and ought to be updated.